lineapy/utils/config.py

```diff
@@ -1,40 +1,181 @@
+
+    def set(self, key: str, value: Any) -> None:
+        """Set LineaPy config field"""
+        if key not in self.__dict__.keys():
+            logger.error(key, "is not a lineapy config item")
+            raise NotImplementedError
+        else:
+            self.__dict__[key] = value
+            os.environ[f"LINEAPY_{key.upper()}"] = str(value)
+
+    def _set_defaults(self):
+        """Fill empty configuration items"""
+        self._safe_get_logging_file()
+        self._safe_get_database_connection_string()
+        self._safe_get_artifact_storage_dir()
+        self._safe_get_customized_annotation_folder()
+
+    def _safe_get_folder(self, name) -> Path:
+        """Return folder as pathlib.Path
+        Create the folder if it doesn't exist"""
+        if not Path(self.__dict__[name]).exists():
+            logger.warning(
+                f"Folder {Path(self.__dict__[name]).as_posix()} does not exist. Creating a new one."
+            )
+            Path(self.__dict__[name]).mkdir(parents=True, exist_ok=True)
+        return Path(self.__dict__[name])
+
+    def _safe_get_logging_file(self) -> Path:
+        """Return logging_file path
+        Use LINEAPY_HOME_DIR/LOG_FILE_NAME if empty"""
+        if self.logging_file is None:
+            self.set(
+                "logging_file",
+                self._safe_get_folder("home_dir").joinpath(LOG_FILE_NAME),
+            )
+        return Path(str(self.logging_file))
+
+    def _safe_get_database_connection_string(self) -> str:
+        """Return LINEAPY_DATABASE_CONNECTION_STRING
+        Use sqlite:///{LINEAPY_HOME_DIR}/{DB_FILE_NAME} if empty"""
+        if self.database_connection_string is None:
+            self.set(
+                "database_connection_string",
+                f"sqlite:///{self._safe_get_folder('home_dir')}/{DB_FILE_NAME}",
+            )
+        return str(self.database_connection_string)
+
+    def _safe_get_artifact_storage_dir(self) -> Path:
+        """Return LINEAPY_ARTIFACT_STORAGE_DIR
+        Use LINEAPY_HOME_DIR/FILE_PICKLER_BASEDIR if empty"""
+        if self.artifact_storage_dir is None:
+            self.set(
+                "artifact_storage_dir",
+                self._safe_get_folder("home_dir").joinpath(
+                    FILE_PICKLER_BASEDIR
+                ),
+            )
+        return self._safe_get_folder("artifact_storage_dir")
+
+    def _safe_get_customized_annotation_folder(self) -> Path:
+        """Return LINEAPY_CUSTOMIZED_ANNOTATION_FOLDER
+        Use LINEAPY_HOME_DIR/CUSTOM_ANNOTATIONS_FOLDER_NAME if empty"""
+        if self.customized_annotation_folder is None:
+            self.set(
+                "customized_annotation_folder",
+                self._safe_get_folder("home_dir").joinpath(
+                    CUSTOM_ANNOTATIONS_FOLDER_NAME
+                ),
+            )
+        return self._safe_get_folder("customized_annotation_folder")
+
+
+options = LineapyConfig()
```

Comment:
Nit: maybe call it `lineapy_config`? a little confusing to call them differently.